data-prepper-plugins/otel-trace-group-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepper.java

```diff
@@ -83,19 +85,20 @@ public OTelTraceGroupPrepper(final PluginSetting pluginSetting) {
             }
         }
 
-        final Map<String, String> traceIdToTraceGroup = searchTraceGroupByTraceIds(traceIdsToLookUp);
+        final Map<String, TraceGroupWrapper> traceIdToTraceGroup = searchTraceGroupByTraceIds(traceIdsToLookUp);
         for (final Map.Entry<Record<String>, Map<String, Object>> entry: recordMissingTraceGroupToRawSpanMap.entrySet()) {
             final Record<String> record = entry.getKey();
             final Map<String, Object> rawSpanMap = entry.getValue();
             final String traceId = (String) rawSpanMap.get(OTelTraceGroupPrepperConfig.TRACE_ID_FIELD);
-            final String traceGroup = traceIdToTraceGroup.get(traceId);
-            if (!Strings.isNullOrEmpty(traceGroup)) {
-                rawSpanMap.put(OTelTraceGroupPrepperConfig.TRACE_GROUP_FIELD, traceGroup);
+            final TraceGroupWrapper traceGroup = traceIdToTraceGroup.get(traceId);
+            if (Objects.nonNull(traceGroup)) {
```

Comment:
Super nit - this kind of threw me off instead of the usual `!= null`.

Might suggest only using the Objects methods when required (like in lambda functions) to keep consistent with the other uses of `!= null`, of which there are 100+.